### ISSUE TITLE
Spiking cell activity plots

### ISSUE THREAD

**Issue by carolinafernandezp** [jonescompneurolab/hnn-core#480]
Hello,

I would like to plot the spiking cell activity from the spk.txt output files.

I have two questions:

**1.** For a total of 10 trials, HNN's output looks like this: spk.txt, spk_0.txt ... spk_9.txt
In this case, what is the "spk.txt" file? Is it some sort of average over the 10 trials?

**2.** The "spk" files consist of two columns: time and cell identifiers. Could you tell me the ranges of the cell identifiers for each type of cell (i.e., layer 2/3 pyramidal, layer 2/3 basket, layer 5 pyramidal, layer 5 basket)? That way I can color them appropriately in my plot.

Thanks in advance,
Carolina

**Comment by rythorpe** [comment]
> **1.** For a total of 10 trials, HNN's output looks like this: spk.txt, spk_0.txt ... spk_9.txt In this case, what is the "spk.txt" file? Is it some sort of average over the 10 trials?

Yes, I believe the "spk.txt" file is some sort of trial average. @ntolley correct me if I'm wrong here....
 
> **2.** The "spk" files consist of two columns: time and cell identifiers. Could you tell me the ranges of the cell identifiers for each type of cell (i.e., layer 2/3 pyramidal, layer 2/3 basket, layer 5 pyramidal, layer 5 basket)? That way I can color them appropriately in my plot.

Check out [this example](https://jonescompneurolab.github.io/hnn-core/stable/auto_examples/howto/plot_firing_pattern.html#sphx-glr-auto-examples-howto-plot-firing-pattern-py). I think it will clarify things for you and provide a straightforward way to plot the spiking activity.

**Comment by ntolley** [comment]
@carolinafernandezp I can't reproduce this on my end. Do you mind sending a minimal example script?

The expected result is that passing the string `fname='spike_%d'` will print out a separate file for each trial. Is it possible you ran the code cell multiple times with different strings?

I'm noticing that when the `%d` is not used, the files are actually overwritten. @rythorpe @jasmainak perhaps this falls under "good first issue" territory to improve the documentation, and raise a warning if the wildcard `%d` is missing when `n_trials > 1`

Lastly the attribute `net.gid_ranges` will give you the cell identifiers.
